batch7crud-roll4/src/main/java/com/lftechnology/batch7crud/dao/EmployeeDao.java

```diff
@@ -0,0 +1,126 @@
+package com.lftechnology.batch7crud.dao;
+
+import com.lftechnology.batch7crud.exception.DataException;
+import com.lftechnology.batch7crud.model.Employee;
+import com.lftechnology.batch7crud.util.DbConnection;
+
+import javax.xml.crypto.Data;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by pratishshr on 1/14/16.
+ */
+public class EmployeeDao {
+    private Connection connection;
+
+    public EmployeeDao() throws DataException {
+        connection = DbConnection.getConnection();
+    }
+
+    public List<Employee> fetch(int page, int recordLimit) throws DataException {
+        try {
+            List<Employee> employeeList = new ArrayList<Employee>();
+            int offset = (page - 1) * recordLimit;
+            String sql = "SELECT * FROM employees LIMIT ? OFFSET ?";
+            PreparedStatement stmt = connection.prepareStatement(sql);
+            stmt.setInt(1, recordLimit);
+            stmt.setInt(2, offset);
+            ResultSet rs = stmt.executeQuery();
+
+            while (rs.next()) {
+                Employee e = new Employee();
+
+                e.setId(rs.getInt("id"));
+                e.setFirstName(rs.getString("first_name"));
+                e.setLastName(rs.getString("last_name"));
+                e.setStation(rs.getString("station"));
+                employeeList.add(e);
+            }
+            return employeeList;
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+    }
+
+    public Employee fetchById(int id) throws DataException {
+        try {
+            Employee e = null;
+            String sql = "SELECT * FROM employees WHERE id=?";
+            PreparedStatement stmt = connection.prepareStatement(sql);
+            stmt.setInt(1, id);
+            ResultSet rs = stmt.executeQuery();
+
+            while (rs.next()) {
+                e = new Employee();
+                e.setId(rs.getInt("id"));
+                e.setFirstName(rs.getString("first_name"));
+                e.setLastName(rs.getString("last_name"));
+                e.setStation(rs.getString("station"));
+            }
+            return e;
+
+        } catch (SQLException e) {
+            throw new DataException();
+        }
+    }
+
+    public void insert(Employee employee) throws DataException {
+        try {
+            String sql = "INSERT INTO employees(first_name, last_name, station) VALUES(?,?,?)";
+            PreparedStatement stmt = connection.prepareStatement(sql);
+            stmt.setString(1, employee.getFirstName());
+            stmt.setString(2, employee.getLastName());
+            stmt.setString(3, employee.getStation());
+
+            stmt.executeUpdate();
+        } catch (SQLException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)